### PR TITLE
Fix MIRROR_IMAGES=False

### PR DESCRIPTION
**Pull request by hjensas**
With https://github.com/openshift-metal3/dev-scripts/pull/1669 the possibility to explicitly set MIRROR_IMAGES=False to disable the implicit MIRROR_IMAGES=True when IP_STACK=v6.

The revert in https://github.com/openshift-metal3/dev-scripts/pull/1694 broke this behaviour because the or logic testing for IP_STACK=v6 result in overriding OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to local mirror even when MIRROR_IMAGES=False was provided by the user.

This changes the logic to use and, checking for MIRROR_IMAGE != False and IP_STACK == v6.

### CODE REVIEW

**Comment by hjensas** [comment]
Looks like ansible-collection-equinix is gone - https://github.com/equinix-labs/ansible-collection-equinix -> `404`

**Comment by andfasano** [comment]
should be back now

/retest

**Comment by andfasano** [comment]
Thanks @hjensas it looks like the fix worked well for the agent jobs (the compact does not have any mirroring)

/approve

**Comment by bfournie** [comment]
Perhaps I need to add something additional to my config_XX.sh with this change. I'm using a local assisted-service repo so mirroring is needed. Its working fine, but when I try this change I get:

```
(./agent/04_agent_prepare_release.sh:49): build_local_release(): echo 'OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry'
OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry
```

**Comment by bfournie** [comment]
> Perhaps I need to add something additional to my config_XX.sh with this change. I'm using a local assisted-service repo so mirroring is needed. Its working fine, but when I try this change I get:
> 
> ```
> (./agent/04_agent_prepare_release.sh:49): build_local_release(): echo 'OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry'
> OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry
> ```



> Perhaps I need to add something additional to my config_XX.sh with this change. I'm using a local assisted-service repo so mirroring is needed. Its working fine, but when I try this change I get:
> 
> ```
> (./agent/04_agent_prepare_release.sh:49): build_local_release(): echo 'OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry'
> OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry
> ```

No, setting
`MIRROR_IMAGES=true`
didn't help, it still failed with above error when using a local repo

**Comment by hjensas** [comment]
> > Perhaps I need to add something additional to my config_XX.sh with this change. I'm using a local assisted-service repo so mirroring is needed. Its working fine, but when I try this change I get:
> > ```
> > (./agent/04_agent_prepare_release.sh:49): build_local_release(): echo 'OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry'
> > OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry
> > ```
> 
> No, setting `MIRROR_IMAGES=true` didn't help, it still failed with above error when using a local repo

Hm, do we need to add something more check's in the `or` block on the right side?

`"${MIRROR_IMAGES,,}" != "true"` or possibly `! -z "${REPO_OVERRIDES}"` is better.
I am guessing REPO_OVERRIDES == buld == must_use_local_mirror ?

`if [[ "${MIRROR_IMAGES,,}" != "false" ]] && [[ "${MIRROR_IMAGES,,}" != "true" -z "${IP_STACK:-}" || "$IP_STACK" = "v6" ]]; then`

or

`if [[ "${MIRROR_IMAGES,,}" != "false" ]] && [[ ! -z "${REPO_OVERRIDES}"  || -z "${IP_STACK:-}" || "$IP_STACK" = "v6" ]]; then`

**Comment by hjensas** [comment]
> > > Perhaps I need to add something additional to my config_XX.sh with this change. I'm using a local assisted-service repo so mirroring is needed. Its working fine, but when I try this change I get:
> > > ```
> > > (./agent/04_agent_prepare_release.sh:49): build_local_release(): echo 'OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry'
> > > OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry
> > > ```
> > 
> > 
> > No, setting `MIRROR_IMAGES=true` didn't help, it still failed with above error when using a local repo
> 
> Hm, do we need to add something more check's in the `or` block on the right side?
> 
> `"${MIRROR_IMAGES,,}" == "true"` or possibly `! -z "${REPO_OVERRIDES}"` is better. I am guessing REPO_OVERRIDES == buld == must_use_local_mirror ?
> 

I added both of these, hopefully that fixes your issue @bfournie ?

**Comment by bfournie** [comment]
> > > > Perhaps I need to add something additional to my config_XX.sh with this change. I'm using a local assisted-service repo so mirroring is needed. Its working fine, but when I try this change I get:
> > > > ```
> > > > (./agent/04_agent_prepare_release.sh:49): build_local_release(): echo 'OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry'
> > > > OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE mismatch, it must reference the local registry
> > > > ```
> > > 
> > > 
> > > No, setting `MIRROR_IMAGES=true` didn't help, it still failed with above error when using a local repo
> > 
> > 
> > Hm, do we need to add something more check's in the `or` block on the right side?
> > `"${MIRROR_IMAGES,,}" == "true"` or possibly `! -z "${REPO_OVERRIDES}"` is better. I am guessing REPO_OVERRIDES == buld == must_use_local_mirror ?
> 
> I added both of these, hopefully that fixes your issue @bfournie ?

Thanks Harald. I tested this and just the addition of `"${MIRROR_IMAGES,,}" == "true"` is enough to resolve this. Its not necessary to also set `! -z "${REPO_OVERRIDES:-}"`

File: common.sh [hunk 1]
```diff
@@ -481,7 +481,7 @@ if [[ ! -z ${AGENT_E2E_TEST_BOOT_MODE} ]]; then
   esac
 fi
 
-if [[ (-n "$MIRROR_IMAGES" && "${MIRROR_IMAGES,,}" != "false") || -z "${IP_STACK:-}" || "$IP_STACK" = "v6" ]]; then
+if [[ "${MIRROR_IMAGES,,}" != "false" ]] && [[ "${MIRROR_IMAGES,,}" == "true" || ! -z "${REPO_OVERRIDES:-}" || -z "${IP_STACK:-}" || "$IP_STACK" = "v6" ]]; then
```
Comment (bfournie) [inline common.sh:484]:
```suggestion
if [[ "${MIRROR_IMAGES,,}" != "false" ]] && [[ "${MIRROR_IMAGES,,}" == "true" || -z "${IP_STACK:-}" || "$IP_STACK" = "v6" ]]; then
```

Comment (hjensas) [reply common.sh]: ok, Done.

**Pull request merged**